Modules/CalcOffence-3_0.lua

```diff
@@ -1172,13 +1172,96 @@ function calcs.offence(env, actor, activeSkill)
 		else
 			output.FistOfWarHitEffect = 1
 			output.FistOfWarAilmentEffect = 1
 		end
 
+		-- Calculate Exerted Attacks
+		output.extraExertions = skillModList:Sum("BASE", nil, "ExtraExertedAttacks") or 0
+		output.exertedUptime = 0
+		local bInstantWarcry = skillModList:Flag(nil, "InstantWarcry") or false
+		
+		output.maxSeismicExerts = skillModList:Sum("BASE", cfg, "SeismicExertedAttacks")
+		output.SeismicHitMultiplier = skillModList:Sum("BASE", cfg, "SeismicHitMultiplier") / 100
+		if output.maxSeismicExerts ~= 0 then
+			output.numSeismicExerts = output.maxSeismicExerts + output.extraExertions
+			for _, value in ipairs(env.auxSkillList) do
+				if value.skillCfg.skillName == "Seismic Cry" then
+					-- get cooldown
+					local gemDefaultCooldown = value.skillData.cooldown
+					output.SeismicCryCooldown = gemDefaultCooldown / (1 + skillModList:Sum("INC", value.skillCfg, "CooldownRecovery") / 100)
+					-- round it to the nearest server tick rate
+					output.SeismicCryCooldown = m_ceil(output.SeismicCryCooldown * data.misc.ServerTickRate) / data.misc.ServerTickRate
+
+					-- get castTime
+					if bInstantWarcry == true then
+						output.SeismicCryCastTime = 0
+					else
+						local gemDefaultCastTime = value.effectList[1].grantedEffect.castTime
+						output.SeismicCryCastTime = gemDefaultCastTime / (1 + skillModList:Sum("INC", value.skillCfg, "WarcrySpeed") / 100)
+						-- round it to the nearest server tick rate
+						output.SeismicCryCastTime = m_ceil(output.SeismicCryCastTime * data.misc.ServerTickRate) / data.misc.ServerTickRate
+					end
+
+					break
+				end
+			end
+			output.SeismicDmgImpact = 0
+			for i = 1, output.numSeismicExerts do
+				output.SeismicDmgImpact = output.SeismicDmgImpact + (i * output.SeismicHitMultiplier)
+			end
+			output.SeismicAvgDmg = output.SeismicDmgImpact / output.numSeismicExerts
+			-- calculate ratio of uptime versus downtime (including opportunity cost of casting the warcry)
+			output.SeismicUpTimeRatio = m_min((output.numSeismicExerts / output.Speed) / (output.SeismicCryCooldown + output.SeismicCryCastTime), 1.0)
+			output.exertedUptime = m_max(output.exertedUptime, output.SeismicUpTimeRatio)
+			output.SeismicHitEffect = 1 + (output.SeismicAvgDmg * output.SeismicUpTimeRatio)
+		else
+			output.SeismicHitEffect = 1
+		end
+
+		output.maxIntimidatingExerts = skillModList:Sum("BASE", nil, "IntimidatingExertedAttacks")
+		if output.maxIntimidatingExerts ~= 0 then
+			exertedAttacks = true
+			output.numIntimidatingExerts = output.maxIntimidatingExerts + output.extraExertions
+			for _, value in ipairs(env.auxSkillList) do
+				if value.skillCfg.skillName == "Intimidating Cry" then
+					local gemDefaultCooldown = value.skillData.cooldown
+					output.IntimidatingCryCooldown = gemDefaultCooldown / (1 + skillModList:Sum("INC", value.skillCfg, "CooldownRecovery") / 100)
```

Comment:
Needs to take into account the base added cooldown of Urgen Orders Support with `skillModList:Sum("BASE", skillCfg, "CooldownRecovery")`